app/services/legacy/dataset_import_service.rb

```diff
@@ -158,11 +164,20 @@ def build_summary(notes)
   end
 
   def build_licence
-    legacy_dataset["license_id"]
+    return 'no-licence' if licence.blank?
+    return 'other' if licence != "uk-ogl"
+    licence
   end
 
   def build_licence_other
-    get_extra('licence')
+    return nil if licence.blank?
+    return licence if licence != "uk-ogl"
+  end
+
+  def build_licence_code
+    code = legacy_dataset["license_id"]
```

Comment:
Rails has http://api.rubyonrails.org/classes/Object.html#method-i-presence which you could maybe use here.

```rb
def build_licence_code
  legacy_dataset["license_id"].presence
end
```

Reply:
You should also use the newly introduced `licence` method. Maybe you could use `presence` in that method?

```rb
def build_licence_code
  licence
end

def licence
  legacy_dataset["license_id"].presence
end
```